### PR TITLE
Discovery Bid Adapter : fix window.top bug

### PR DESCRIPTION
**Pull request by lhxx121**
## Type of change

- [x] Bugfix
- [ ] Feature
- [ ] New bidder adapter  
- [ ] Updated bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes

- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change

Fix bugs caused by Same-Origin Policy

### CODE REVIEW

**Comment by lhxx121** [comment]
@ChrisHuie Hi, sorry to bother you, could you please help me cr it? This submission is very important to us.❤️

**Review by patmmccann — CHANGES_REQUESTED** [review]

**Comment by Rothalack** [comment]
The code changes look good, but I am unable to checkout this pr and test. I think you need to rebase master. It doesn't make sense to me why, these changes shouldn't conflict, but I don't want to try to rebase or change anything on my end to get this working.

File: modules/discoveryBidAdapter.js [hunk 1]
```diff
@@ -116,6 +116,36 @@ export function getPageKeywords(win = window) {
...
+  } catch (error) {
+    hc = undefined;
+  }
+  return hc;
+}
+
+export function getDM(win = window) {
+  let dm;
+  try {
+    dm = win.top.navigator.deviceMemory;
```
[the remark this answers is not in the document]
Comment (lhxx121) [reply modules/discoveryBidAdapter.js]: I understand. I will modify and submit immediately

**Comment by lhxx121** [comment]
@Rothalack I understand. I will modify and submit immediately

**Review by Rothalack — APPROVED** [review]
I still had to manually check this out into a new branch for myself to test. Everything is working in testing. I still don't know why github complains about this when I try to gh pr checkout. I think it's because there's some divergence somewhere in your commit history that makes the simple gh checkout not work. I don't see any reason why that divergence is going to cause any issues. There's no conflicts. So I think we can just ignore it and merging this will work fine.

**Review by patmmccann — APPROVED** [review]

**Pull request merged**